agent/config/parse_windows_test.go

```diff
@@ -50,3 +55,143 @@ func TestParseFSxWindowsFileServerCapability(t *testing.T) {
 
 	assert.False(t, parseFSxWindowsFileServerCapability())
 }
+
+func TestGetOperatingSystemFamilyForWS2019Core(t *testing.T) {
```

Comment:
To avoid mostly repeated code, you can use test tables. Here's an example of how to do it:

https://github.com/aws/amazon-ecs-agent/blob/2874131a5f3d1c0908e0f2ebad62e48b8ff5850f/agent/engine/execcmd/agent_version_test.go#L146

Reply:
I tried to make tests better by creating a function for common code. However the test tables doesn't seem a good idea here , I tried it and it looks bad to me. Here are the following reasons.
1.  I see the go test tables are nothing but an array of structure defining inputs , I don't see this mechanism reducing the LOC at least in this case.
2. The tests here have different expectations and not possible to use just one for loop to run all tests. I can put multiple but it didn't look great to me.
3. These are different test cases , its hard to debug individual tests without commenting others.

I see your concern regarding redundant code and I agree , to address it I have moved most of the redundant code to functions.  If i use task tables it doesn't really good to me ( i tried it Here is the link if you wanted to look: https://paste.amazon.com/show/sssandee/1620245472